src/features/copy-paste-menu-selection-override.js

```diff
@@ -0,0 +1,215 @@
+                }
+
+                // helper function to enumerate all text nodes in selection range
+                function forEachTextNodeInRange(range, callback) {
+                    const startContainer = range.startContainer;
+                    const endContainer = range.endContainer;
+
+                    // helper function to enumerate all text nodes in a node
+                    function enumerateTextNodes(node, callback) {
+                        if (node.nodeType === Node.TEXT_NODE) {
+                            return callback(node);
+                        } else {
+                            for (let child = node.firstChild; child; child = child.nextSibling) {
+                                if (enumerateTextNodes(child, callback) === false) {
+                                    return false;
+                                }
+                            }
+                        }
+                    }
+
+                    // get the common ancestor container of the range
+                    const commonAncestorContainer = range.commonAncestorContainer;
+                    // track whether the enumeration has got into the selection range
+                    let inRange = false;
+
+                    enumerateTextNodes(commonAncestorContainer, function(node) {
+                        if (node === startContainer) {
+                            inRange = true;
+                        }
+                        // stop enumeration when the selection range end node is reached
+                        var shouldContinue = !(node === endContainer);
+                        if (inRange) {
+                            shouldContinue = callback(node) && shouldContinue;
+                        }
+
+                        return shouldContinue;
+                    });
+                }
+
+                // - Override transparent text selection color resetting it to default selection color
+                document.addEventListener('selectionchange', function(e) {
+                    const selection = document.getSelection();
+                    if (selection.rangeCount === 0) { return }
+                    const range = selection.getRangeAt(0);
+
+                    // enumerate all text nodes in selection
+                    forEachTextNodeInRange(range, ((text) => {
+                        const node = text.parentNode;
+                        if (node.classList.contains('__ddg-override-selection-background-color')) {
+                            return true; // selection color is already overriden – proceed next
+                        }
+
+                        const selectionColor = window.getComputedStyle(node, '::selection').backgroundColor;
+                        // if text node selection color is set to `transparent` (0,0,0,0) or is not set explicitly - set it to default (`highlight`)
+                        if (selectionColor === 'rgba(0, 0, 0, 0)' || selectionColor === 'transparent') {
```

Comment:
There's a bunch of other values here that are could cause it not to show. Like visibility:hidden, also does #00000000 appear in computed styles or is it translated to rgba?

Reply:
It seems like the default in Chrome is `rgba(0, 0, 0, 0)`, I'm not sure how the style applies but we should probably not apply in default case.

Reply:
yes, the default is detected `rgba(0, 0, 0, 0)` (i.e. not explicitly defined), but may also be used from the client side to hide selection.

>  but we should probably not apply in default case

I agree, but there seems no way to distinguish the "not defined" case and an explicit `rgba(0, 0, 0, 0)`. The overridden style sets selection background color to `highlight` which is a default browser selection color, so in case it‘s not explicitly defined – the selection colors remains the same.